admin/admin.go

```diff
@@ -260,16 +261,18 @@ func StartServer(ctx context.Context) error {
 		pkgLogger.Fatal("listen error:", e) // @TODO return?
 	}
 	defer func() {
-		if err := l.Close(); err != nil {
-			pkgLogger.Warn(err)
+		if l != nil {
+			if err := l.Close(); err != nil {
+				pkgLogger.Warn(err)
+			}
 		}
 	}()
 
 	pkgLogger.Info("Serving on admin interface @ ", sockAddr)
 	srvMux := http.NewServeMux()
 	srvMux.Handle(rpc.DefaultRPCPath, instance.rpcServer)
 
-	srv := &http.Server{Handler: srvMux}
+	srv := &http.Server{Handler: srvMux, ReadHeaderTimeout: 3 * time.Second}
```

Comment:
deepsource was complaining about some attack vector, I think this won't hurt anyway here :)